### PR TITLE
DSND-2473: Implement GET company filing history list endpoint

### PR DESCRIPTION
**Pull request by cgregoryCH**
## Describe the changes
* Endpoint returns a list of filing history transaction for a given company number.
* Includes sorting, pagination and filtering on category.
* Required various changes to the private spec which are raised in [private.api.ch.gov.uk-specifications #268](https://github.com/companieshouse/private.api.ch.gov.uk-specifications/pull/268)

### Related Jira tickets

[DSND-2473](https://companieshouse.atlassian.net/browse/DSND-2473)

### General

- [x] Is the PR as small as it can be?
- [x] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [x] Has the `POM` been updated for the latest versions of dependencies?
- [x] Has the code been double-checked against `main`?
- [x] Does the code adhere standards in this repository, including SonarQube checks?

### Testing

- [x] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation

_Where possible, add links to the respective Jira tickets when an item is checked_

- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to deployment repositories?

## Notes to the tester

_Add any testing hints or instructions here._

[DSND-2473]: https://companieshouse.atlassian.net/browse/DSND-2473?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/filinghistory/api/service/Service.java [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
 import java.util.Optional;
-import uk.gov.companieshouse.filinghistory.api.model.FilingHistoryDocument;
+import uk.gov.companieshouse.filinghistory.api.model.mongo.FilingHistoryDocument;
+import uk.gov.companieshouse.filinghistory.api.model.mongo.FilingHistoryListAggregate;
 
 public interface Service {
 
+    Optional<FilingHistoryListAggregate> findCompanyFilingHistoryList(String companyNumber, int startIndex,
+            int itemsPerPage, List<String> categories);
+
```
Comment (crowlandsCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/api/service/Service.java:12]: bunch of whitespace lines in here between the methods we could remove unless people think it impacts readability to do so.

**Review by crowlandsCH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/filinghistory/api/service/Service.java (hunk 1 above)
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/filinghistory/api/service/Service.java:12]: fair enough, I see you formatted the other interface methods to be consistent nice.

**Review by kinpang-CH — APPROVED** [review]

**Pull request merged**